cpp_interface.dd

```diff
@@ -296,328 +340,406 @@ class E
+
+    $(P C++ and D follow different rules for function overloading.
+    D source code, even when calling $(CODE extern (C++)) functions,
+    will still follow D overloading rules.
+    )
+
+$(H2 $(LNAME2 memory-allocation, Memory Allocation))
+
+    $(P C++ code explicitly manages memory with calls to
+    $(CODE ::operator new()) and $(CODE ::operator delete()).
+    D's $(CODE new) operator allocates memory using the D garbage collector,
+    so no explicit delete is necessary. D's $(CODE new) operator is not
+    compatible with C++'s $(CODE ::operator new) and $(CODE::operator delete).
+    Attempting to allocate memory with D's $(CODE new) and deallocate with
+    C++ $(CODE ::operator delete) will result in miserable failure.
+    )
+
+    $(P D can explicitly manage memory using a variety of library tools, such as
+    with $(STDMODREF2 experimental, allocator, std.allocator). Additionally,
+    $(CODE core.stdc.stdlib.malloc) and $(CODE core.stdc.stdlib.free) can be
+    used directly for connecting to C++ functions that expect $(CODE malloc)'d
+    buffers.
+    )
+
+    $(P If pointers to memory allocated on the D garbage collector heap are
+    passed to C++ functions, it's critical to ensure that the referenced memory
+    will not be collected by D the garbage collector before the C++ function is
+    done with it. This is accomplished by:
+    )
+
+    $(UL
+
+    $(LI Making a copy of the data using
+    $(STDMODREF2 experimental, allocator, std.allocator) or
+    $(CODE core.stdc.stdlib.malloc) and passing the copy instead.)
+
+    $(LI Leaving a pointer to it on the stack (as a parameter or
+    automatic variable), as the garbage collector will scan the stack.)
+
+    $(LI Leaving a pointer to it in the static data segment, as the
+    garbage collector will scan the static data segment.)
+
+    $(LI Registering the pointer with the garbage collector using the
+    $(CODE core.memory.GC.addRoot) or $(CODE core.memory.GC.addRange)
+    functions.)
+
+    )
+
+    $(P An interior pointer to the allocated memory block is sufficient to let
+    the GC know the object is in use; i.e. it is not necessary to maintain
+    a pointer to the $(I beginning) of the allocated memory.
+    )
+
+    $(P The garbage collector does not scan the stacks of threads not
+    registered with the D runtime, nor does it scan the data segments of
+    shared libraries that aren't registered with the D runtime.
+    )
+
+$(H2 $(LNAME2 data-type-compatibility, Data Type Compatibility))
+
+    $(TABLE2 D And C++ Type Equivalence,
+
+    $(THEAD D type, C++ type)
+
+    $(TROW
+    $(ARGS $(B void)),
+    $(ARGS $(B void))
+    )
+
+    $(TROW
+    $(ARGS $(B byte)),
+    $(ARGS $(B signed char))
+    )
+
+    $(TROW
+    $(ARGS $(B ubyte)),
+    $(ARGS $(B unsigned char))
+    )
+
+    $(TROW
+    $(ARGS $(B char)),
+    $(ARGS $(B char) (chars are unsigned in D))
+    )
+
+    $(TROW
+    $(ARGS $(B core.stdc.stddef.wchar_t)),
```

Comment:
Or use `$(D ...)`, as is more appropriate here because you're referring to an identifier.